File: wlr/OGUR/OGUR/Text/DefaultHudText.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using OGUR.Creatures;
using OGUR.Management;

namespace OGUR.Text
{
    class DefaultHudText:Text
    {
        private Color m_color;
        private Vector2 m_origin;

        public DefaultHudText(float alpha=1f)
        {
            m_color = new Color(255f, 255f, 255f, alpha);
        }

        public DefaultHudText(string contents, int x, int y,Vector2 origin,float alpha=1f)
            : base(contents, x, y,TextType.Inventory)
        {
            m_color = new Color(255f,255f,255f,alpha);
            m_origin = origin;
        }

        public void Reset(string contents, int x, int y,Vector2 origin)
        {
            m_origin = origin;
            Reset(contents, x, y);
        }

        public override int Update()
        {
           return 1;
        }
        public override void Draw()
        {
            XnaManager.Renderer.DrawString(TextManager.GetFont(), m_contents, m_position+m_origin, m_color, 0, Vector2.Zero,.75f, SpriteEffects.None, .96f);
        }
    }
}

File: aigilas/src/sps/text/DefaultHudText.java
package sps.text;

import com.badlogic.gdx.graphics.Color;
import sps.bridge.DrawDepth;
import sps.core.Point2;
import sps.core.Settings;
import sps.core.SpxManager;

class DefaultHudText extends Text {
    private final Color _color;
    private Point2 _origin;

    public DefaultHudText(float alpha) {
        _color = new Color(255f, 255f, 255f, alpha);
    }

    public void reset(String contents, int x, int y, Point2 origin) {
        _origin = origin;
        reset(contents, x, y);
    }

    @Override
    public int update() {
        return 1;
    }

    @Override
    public void draw() {
        SpxManager.Renderer.drawString(_contents, new Point2(_position.X + Settings.get().spriteWidth, SpxManager.VirtualHeight - _position.Y + Settings.get().spriteHeight / 2), _color, 1f, DrawDepth.DefaultHudText);
    }
}
